### PR TITLE
Merges 2.10.x

### PR DESCRIPTION
**Pull request by jsuereth**
Fixes broken tests and migrates a few new reclection methods to the time-travel refactoring.

Review by @paulp.

### CODE REVIEW

File: src/actors-migration/scala/actors/migration/Pattern.scala [hunk 1]
```diff
@@ -2,11 +2,11 @@ package scala.actors.migration
 
 import scala.actors._
 import scala.concurrent.duration.Duration
-import scala.language.implicitConversions
+import language.implicitConversions
```
Comment (paulp) [inline src/actors-migration/scala/actors/migration/Pattern.scala:6]: Already starting to undo all my good works, huh? No relative imports!

**Comment by paulp** [comment]
Other than that one line it looks good to me, although if there was an issue in there somewhere I'd be hard pressed to spot it.

File: src/actors-migration/scala/actors/migration/Pattern.scala (hunk 1 above)
Comment (jsuereth) [inline src/actors-migration/scala/actors/migration/Pattern.scala:6]: Wow.  Yeah, github merging.  Did you have something that spews pokemon warnings for this so I can catch all of em?

**Comment by jsuereth** [comment]
If it helps:  The time-traveling methods caused the only real merge conflict.   Some new tests in 2.10.x were using afterPhase which I migrated to endingPhase or whatever based on the commit message that fixed stuff.

**Comment by paulp** [comment]
> If it helps: The time-traveling methods caused the only real merge conflict.

Those merge conflicts have been a constant companion since I committed them to master, which I did only due to the absence of realism about the 2.10.x branch.  An anecdote I offer in case anyone thinks there's no cost associated with these things.

**Comment by gkossakowski** [comment]
@paulp: So we ready to go with this one?

**Comment by paulp** [comment]
Yeah, the import he ruined is no big deal, I'll fix it later.

**Pull request merged**